### ISSUE TITLE
Bug: jsonc/indent does not report extra spaces after a tab

### ISSUE THREAD

**Issue by Cherepanov-Pavel** [ota-meshi/eslint-plugin-jsonc#543]
First of all, thank you very much for this great and very useful package! eslint-plugin-jsonc makes it much easier to apply consistent ESLint rules to JSON, JSONC, and JSON5 files.

I found what seems to be a bug in the jsonc/indent rule when using tabs for indentation.

### Configuration
For JavaScript/TypeScript files, I use @stylistic/indent with the following configuration:
`'@stylistic/indent': [ 'error', 'tab', { tabLength: 2 }]`

For JSON files, I use:
`'jsonc/indent': [ 'error', 'tab' ]`

### Expected behavior
Consider this valid TypeScript code:
```
const obj = {
	test: 3,
};
```
The test property is indented by one tab.
If I add spaces after the tab:
```
const obj = {
						test: 3,
};
```
@stylistic/indent correctly reports an error:

> Expected indentation of 1 tab but found 23 spaces.

I would expect jsonc/indent to behave in the same way for JSON.

### Actual behavior
For the following JSON:
```
{
						"test": 3
}
```

jsonc/indent does not report any error, even though the indentation contains a tab followed by additional spaces and therefore does not match the configured tab indentation style. Interestingly, indentation consisting only of tabs is handled correctly. If there are more or fewer tabs than expected, jsonc/indent correctly reports an indentation error.
### Expected result
I would expect jsonc/indent with:
`'jsonc/indent': ['error', 'tab']`
to reject indentation that contains extra spaces after a tab, similarly to @stylistic/indent.

Is this behavior expected, or could this be considered a bug in jsonc/indent?

Thank you again for maintaining such a useful package!
